lib/backpex/html/form.ex

```diff
@@ -4,152 +4,249 @@ defmodule Backpex.HTML.Form do
+          <input
+            type="checkbox"
+            id={@id}
+            name={@name}
+            value="true"
+            checked={@checked}
+            class={[
+              "checkbox checkbox-sm phx-no-feedback:checkbox phx-no-feedback:checkbox-primary",
+              @input_class,
+              @errors == [] && "checkbox-primary",
+              @errors != [] && "checkbox-error"
+            ]}
+            {@rest}
+          />
+        <% end %>
+      </div>
+      <.error :for={msg <- @errors} :if={not @hide_errors}><%= msg %></.error>
     </div>
     """
   end
 
-  def field_input(%{type: "textarea"} = assigns) do
-    assigns =
-      assigns
-      |> assign_new(:errors, fn -> Keyword.get_values(assigns.form.errors || [], assigns.field_name) end)
+  def field_input(%{type: "toggle", value: value} = assigns) do
+    assigns = assign_new(assigns, :checked, fn -> Form.normalize_value("checkbox", value) end)
 
     ~H"""
-    <div phx-feedback-for={PhoenixForm.input_name(@form, @field_name)}>
-      <%= PhoenixForm.textarea(
-        @form,
-        @field_name,
-        [
-          class: [
-            "phx-no-feedback:textarea phx-no-feedback:textarea-bordered w-full",
-            @errors == [] && "textarea textarea-bordered",
-            @errors != [] && "textarea textarea-error bg-red-50 placeholder-danger"
-          ],
-          placeholder: Backpex.Field.placeholder(@field_options, assigns),
-          phx_debounce: Backpex.Field.debounce(@field_options, assigns),
-          phx_throttle: Backpex.Field.throttle(@field_options, assigns)
-        ] ++ Map.to_list(@rest)
-      ) %>
-      <.error_tag form={@form} name={@field_name} field_options={@field_options} />
+    <div phx-feedback-for={@name} class={@class}>
+      <div class="form-control w-full">
+        <%= if @label do %>
+          <label class="label cursor-pointer">
+            <input type="hidden" name={@name} value="false" />
+            <input
+              type="checkbox"
+              id={@id}
+              name={@name}
+              value="true"
+              checked={@checked}
+              class={[
+                "toggle phx-no-feedback:toggle phx-no-feedback:toggle-primary",
+                @input_class,
+                @errors == [] && "toggle-primary",
+                @errors != [] && "toggle-error"
+              ]}
+              {@rest}
+            />
+            <span class="label-text ml-2"><%= @label %></span>
+          </label>
+        <% else %>
+          <input type="hidden" name={@name} value="false" />
+          <input
+            type="checkbox"
+            id={@id}
+            name={@name}
+            value="true"
+            checked={@checked}
+            class={[
+              "toggle phx-no-feedback:toggle phx-no-feedback:toggle-primary",
+              @input_class,
+              @errors == [] && "toggle-primary",
+              @errors != [] && "toggle-error"
+            ]}
```

Comment:
Maybe add a helper for this common case?

- first param: default class
- second param: no error classes
- third param: error classes

```suggestion
            class={input_classes(["toggle phx-no-feedback:toggle phx-no-feedback:toggle-primary", @input_class], "toggle-primary", "toggle-error")}
```

Reply:
In this case, would we also need to pass the `@errors` attribute to the `input_classes` function, or how does it decide whether to include error classes or no error classes?

Reply:
Right... maybe just leave it as is.